cdap-data-fabric/src/main/java/co/cask/cdap/data2/transaction/Transactions.java

```diff
@@ -344,4 +348,59 @@ public static RuntimeException propagate(TransactionFailureException e) {
   private Transactions() {
     // Private the constructor for util class
   }
+
+  /**
+   * Determines the transaction control of a method, as (optionally) annotated with @TransactionPolicy.
+   * If the program's class does not implement the method, the superclass is inspected up to and including
+   * the given base class.
+   *
+   * @param defaultValue returned if no annotation is present
+   * @param baseClass upper bound for the super classes to be inspected
+   * @param program the program
+   * @param methodName the name of the method
+   * @param params the parameter types of the method
+   *
+   * @return the transaction control annotated for the method of the program class or its nearest superclass that
+   *         is annotated; or defaultValue if no annotation is present in any of the superclasses
+   */
+  public static TransactionControl getTransactionControl(TransactionControl defaultValue, Class<?> baseClass,
+                                                         Object program, String methodName, Class<?>... params) {
+    Class<?> cls = program.getClass();
+    while (true) {
+      TransactionControl txControl = getTransactionControl(cls, methodName, params);
+      if (txControl != null) {
+        return txControl;
+      }
+      if (cls == baseClass) {
+        break; // reached bounding superclass
+      }
+      cls = cls.getSuperclass();
+      if (cls == null) {
```

Comment:
We can stop at `Object.class` instead of the parent of `Object.class` (i.e. null), right?